tools/scripts-config-react-webpack/config/webpack.config.js

```diff
@@ -93,28 +96,92 @@ function getSassLoaders(enableModules, sassData, mode) {
 	});
 }
 
-function getVersions() {
-	const talendLibraries = cdn
-		.getModulesFromLockFile()
-		.filter(Boolean)
-		.map(info => {
-			return { version: info.version, name: info.name };
-		});
-	const packageJson = require(path.join(process.cwd(), 'package.json'));
+const BLOCKED_LIST = ['@talend/backend-mock', '@talend/html-webpack-plugin', '@talend/copylib'];
+const BLOCKED_SUBSTRING = ['scripts', 'webpack'];
 
+function isBlockedLib(name) {
+	if (BLOCKED_LIST.indexOf(name) !== -1) {
+		return true;
+	}
+	let blocked = false;
+	BLOCKED_SUBSTRING.forEach(pattern => {
+		if (!blocked) {
+			blocked = name.includes(pattern);
+		}
+	});
+	return blocked;
+}
+
+function getGitRevision() {
 	let revision = process.env.GIT_COMMIT;
 	if (!revision) {
 		try {
 			revision = childProcess.execSync('git rev-parse HEAD').toString().trim();
 		} catch (e) {
+			// eslint-disable-next-line no-console
 			console.info('Failed to get git revision');
 		}
 	}
+	return revision;
+}
+
+function getTalendVersions() {
+	const talendLibraries = {};
+	const packagelockPath = path.join(process.cwd(), 'package-lock.json');
+	const yarnlockPath = path.join(process.cwd(), 'yarn.lock');
+	// eslint-disable-next-line
+	const packageJson = require(path.join(process.cwd(), 'package.json'));
+
+	if (fs.existsSync(yarnlockPath)) {
+		const data = fs.readFileSync(yarnlockPath, 'utf-8');
+		const lock = yarnlock.parse(data);
+		Object.keys(lock.object)
+			.filter(k => k.startsWith('@talend'))
+			.reduce((acc, key) => {
+				// @talend/react-components@5.1.2
+				const name = `@talend/${key.split('/')[1].split('@')[0]}`;
+				const info = lock.object[key];
+				if (!isBlockedLib(name)) {
+					acc[name] = info.version;
+				}
+				return acc;
+			}, talendLibraries);
+	} else if (fs.existsSync(packagelockPath)) {
+		// eslint-disable-next-line
+		const packageLock = require(packagelockPath);
+		Object.keys(packageLock.dependencies)
+			.filter(k => k.startsWith('@talend'))
+			.reduce((acc, key) => {
```

Comment:
this won't work with npm > 6 (which is used on TDP)
also, it's weird webpack doesn't have a plugin for this 🤔

Reply:
Thanks for the feedback 👍🏼 
Is it ok to only check package-lock for npm 6+ or do i have to handle the version before ?

Reply:
I guess you would be better off getting versions from `node_modules/@talend/**/package.json`